Type 'typeof AxisModifier' does not satisfy the constraint 'PluginConstructor<any, any, PluginOptions>'.

Me:

```
"dependencies": {
    "@dnd-kit/helpers": "^0.4.0",
    "@dnd-kit/react": "^0.4.0",
    ...
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
},
"devDependencies": {
    ...
    "ts-node": "^10.9.2",
    "typescript": "6.0.3",
    ...
}
```

The compiler:

```
node_modules/@dnd-kit/abstract/modifiers.d.ts:39:92 - error TS2344: Type 'typeof AxisModifier' does not satisfy the constraint 'PluginConstructor<any, any, PluginOptions>'.
Types of construct signatures are incompatible.
Type 'new (manager: DragDropManager<any, any>, options?: Options$1 | undefined) => AxisModifier' is not assignable to type 'new (manager: any, options?: PluginOptions | undefined) => any'.
Types of parameters 'options' and 'options' are incompatible.
Type 'PluginOptions | undefined' is not assignable to type 'Options$1 | undefined'.
Type 'PluginOptions' is missing the following properties from type 'Options$1': axis, value

39     static configure: (options: Options$1) => _dnd_kit_abstract.PluginDescriptor<any, any, typeof AxisModifier>;
~~~~~~~~~~~~~~~~~~~

node_modules/@dnd-kit/abstract/modifiers.d.ts:47:84 - error TS2344: Type 'typeof AxisModifier' does not satisfy the constraint 'PluginConstructor<any, any, PluginOptions>'.
Types of construct signatures are incompatible.
Type 'new (manager: DragDropManager<any, any>, options?: Options$1 | undefined) => AxisModifier' is not assignable to type 'new (manager: any, options?: PluginOptions | undefined) => any'.
Types of parameters 'options' and 'options' are incompatible.
Type 'PluginOptions | undefined' is not assignable to type 'Options$1 | undefined'.
Type 'PluginOptions' is missing the following properties from type 'Options$1': axis, value

47 declare const RestrictToVerticalAxis: _dnd_kit_abstract.PluginDescriptor<any, any, typeof AxisModifier>;
~~~~~~~~~~~~~~~~~~~

node_modules/@dnd-kit/abstract/modifiers.d.ts:54:86 - error TS2344: Type 'typeof AxisModifier' does not satisfy the constraint 'PluginConstructor<any, any, PluginOptions>'.
Types of construct signatures are incompatible.
Type 'new (manager: DragDropManager<any, any>, options?: Options$1 | undefined) => AxisModifier' is not assignable to type 'new (manager: any, options?: PluginOptions | undefined) => any'.
Types of parameters 'options' and 'options' are incompatible.
Type 'PluginOptions | undefined' is not assignable to type 'Options$1 | undefined'.
Type 'PluginOptions' is missing the following properties from type 'Options$1': axis, value

54 declare const RestrictToHorizontalAxis: _dnd_kit_abstract.PluginDescriptor<any, any, typeof AxisModifier>;
~~~~~~~~~~~~~~~~~~~

node_modules/@dnd-kit/abstract/modifiers.d.ts:131:90 - error TS2344: Type 'typeof SnapModifier' does not satisfy the constraint 'PluginConstructor<any, any, PluginOptions>'.
Types of construct signatures are incompatible.
Type 'new (manager: DragDropManager<any, any>, options?: Options | undefined) => SnapModifier' is not assignable to type 'new (manager: any, options?: PluginOptions | undefined) => any'.
Types of parameters 'options' and 'options' are incompatible.
Type 'PluginOptions | undefined' is not assignable to type 'Options | undefined'.
Property 'size' is missing in type 'PluginOptions' but required in type 'Options'.

131     static configure: (options: Options) => _dnd_kit_abstract.PluginDescriptor<any, any, typeof SnapModifier>;
~~~~~~~~~~~~~~~~~~~

node_modules/@dnd-kit/abstract/modifiers.d.ts:91:5
91     size: number | {
~~~~
'size' is declared here.
```

--- a/.changeset/bold-eyes-start.md
+++ b/.changeset/bold-eyes-start.md
@@ -1,0 +1,5 @@
+---
+'@dnd-kit/abstract': patch
+---
+
+Supply correct Options type in return type of plugin configurator()

--- a/packages/abstract/src/core/plugins/utilities.ts
+++ b/packages/abstract/src/core/plugins/utilities.ts
@@ -1,3 +1,4 @@
+import type {Plugin} from './plugin.ts';
 import type {
   PluginConstructor,
   PluginOptions,
@@ -34,7 +35,7 @@ export function configure<
 export function configurator<T extends PluginConstructor<any, any, any>>(
   plugin: T
 ) {
-  return (options: InferPluginOptions<T>): PluginDescriptor<any, any, T> => {
+  return (options: InferPluginOptions<T>): PluginDescriptor<any, Plugin<any, InferPluginOptions<T>>, T> => {
     return configure(plugin, options);
   };
 }

